config/crds/cdap_v1alpha1_cdapmaster.yaml

```diff
@@ -197,6 +197,12 @@ spec:
                     the service.
                   type: string
               type: object
+            queuedImage:
```

Comment:
Why need a new field? How does it relate to the image field? After the upgrade will this goes retained? Will the image field get changed?

For other controller, typically upgrade is done through updating of the image field

Reply:
The image field gets changed by the operator once the job finishes. Reason for having this as a separate field is that the clh will query the CR, so if we update the image then there's a race condition between querying the CR and having the operator set the image back to the old version while waiting for the job to run.

Reply:
Anything that is being changed by the operator and query by external entities should be under status, not under spec.

Reply:
Done